### DISCUSSION TITLE
Divide mesh according to level of detail

### DISCUSSION THREAD

**Discussion by Megidd** [libigl/libigl#1742] [Q&A] [answered]
### Background

I need to remesh by OpenVDB, like what [Blender remesh](https://docs.blender.org/manual/en/dev/modeling/modifiers/generate/remesh.html) does:

[![Blender screenshot][1]][1]

> Voxel
>
> Uses an OpenVDB to generate a new manifold mesh from the current geometry while trying to preserve the mesh’s original volume.

### Problem

The problem is that OpenVDB gets a constant *voxel size*. But I have meshes with varying polygon density/granularity, i.e. high-detail and low-detail components. To preserve the topology of high-detail components, the *voxel size* has to be too small for the whole 3d model.

[![Polygon granularity/density][2]][2]

### Possible workaround

A mesh might be broken down into distinct components of higher detail and lower detail.

### Question

Is there any tool which might help me break down the mesh into distinct components according to level of detail? 

  [1]: https://i.stack.imgur.com/CJikC.png
  [2]: https://i.stack.imgur.com/jdt0F.png

**Comment by jdumas** [comment, marked as answer]
OpenVDB might not be the best tool for this job. Other remeshing algorithms that do not rely on voxelization might be more adapted. For example:
- [fTetWild](https://github.com/wildmeshing/fTetWild) if your mesh has topological defects.
- [mmgs](https://www.mmgtools.org/), if your mesh is clean and has sharp features.
- [geogram](http://alice.loria.fr/software/geogram/doc/html/index.html)/[graphite](http://alice.loria.fr/software/graphite/doc/html/) is another fast Voronoi-based remesher.
- Libigl has some functions for mesh decimation (tutorial 703), and to remesh self-intersections (in the CGAL subfolder), but this is not the same as a full fledge remesher.
